packages/neon/neon/lib/src/widgets/relative_time.dart

```diff
@@ -10,40 +13,33 @@ class RelativeTime extends StatelessWidget {
   final DateTime date;
   final TextStyle? style;
 
-  static String format(final DateTime date) {
-    final now = DateTime.now();
-    var diff = now.difference(date.toLocal());
-    final text = StringBuffer();
-
-    // Sometimes something can be messed up...
-    if (diff.isNegative) {
-      if (diff.inMinutes >= 1) {
-        text.write('-');
-      }
-      diff = Duration(microseconds: -diff.inMicroseconds);
-    }
-
-    if (diff.inMinutes < 1) {
-      text.write('now');
-    } else if (diff.inHours < 1) {
-      text.write('${diff.inMinutes}m');
-    } else if (diff.inDays < 1) {
-      text.write('${diff.inHours}h');
-    } else if (diff.inDays < 365) {
-      text.write('${diff.inDays}d');
-    } else {
-      text.write('${diff.inDays ~/ 365}y');
-    }
-
-    return text.toString();
+  @override
+  State<RelativeTime> createState() => _RelativeTimeState();
+}
+
+class _RelativeTimeState extends State<RelativeTime> {
+  late final Timer timer;
+
+  @override
+  void initState() {
+    timer = Timer.periodic(
```

Comment:
I like the stream version more. Is there a particular reason you chose to use the timer?

Reply:
There isn't any benefit.